### PR TITLE
Misc changes

### PR DESCRIPTION
**Pull request by bep**
* Add a new WithCodeBlockOptions that allows setting Chroma options per code block.
  * Make linenos attribute Pygments compatible. It can be both booleans and strings (table/inline).
* Fixes a bug where the shared FormatOptions was appended to. Make a copy first.

### CODE REVIEW

**Comment by bep** [comment]
OK; I think this should make sense now. It still needs the Chroma PR and could probably do with a 
squash. 

One question: Is this superfluous? I don't see it getting invoked:

https://github.com/yuin/goldmark-highlighting/blob/631f3f291c26dd1ae887ff4d07198255d49d05ab/highlighting.go#L149

**Comment by yuin** [comment]
You need to implement `SetOption` . `SetOption` is called from [here](https://github.com/yuin/goldmark/blob/master/renderer/renderer.go#L122).

**Comment by bep** [comment]
I have updated the first comment with the changes made and squashed the commits.

So, this now works for me, but it should be improved to make it easier to use, ref.:

https://github.com/yuin/goldmark-highlighting/blob/34ab6e256c83f80b202f6d39cefd9de5ed5e1a14/highlighting.go#L465

The above logic will break if you also enable line numbers with `WithFormatOptions`.

Maybe depending on the outcome of https://github.com/alecthomas/chroma/issues/301 -- but I suggest you change `WithFormatOptions` to take a custom struct that allows toggling of options.

I can make this change as part of this PR (this will then make the `WithLineNumbers` and `WithLineNumbersInTable` superfluous) if you want, or you/we could do it later.

**Comment by bep** [comment]
>I can make this change as part of this PR (this will then make the WithLineNumbers and WithLineNumbersInTable superfluous) if you want, or you/we could do it later.

Second thought, I'll append this as a separate commit to this PR, and you can have a look. It will greatly simplify both the code and the API, I think.

**Comment by bep** [comment]
>I can make this change as part of this PR (this will then make the WithLineNumbers and WithLineNumbersInTable superfluous) if you want, or you/we could do it later.

Second thought, that would make a little bit too much duplication. I suggest we keep it as in this PR for now and you can consider changing it if the Chroma issue somehow gets accepted.

**Comment by bep** [comment]
Seems that https://github.com/alecthomas/chroma/issues/301 is going to be accepted. I will fix that first, which should make this plenty simpler.

**Comment by bep** [comment]
@yuin Chroma has tagged a new release, so this PR is ready to be considered for a merge. See the first comment for a list of changes.

**Comment by yuin** [comment]
@bep I take a quick look of this, it seems many unnecessary dependencies has been added to the `go.mod` . Could you give a try to `go mod tidy?` .

**Comment by bep** [comment]
>Could you give a try to go mod tidy? 

Just did. 

I originally did:

```bash
go get -u github.com/alecthomas/chroma
```

To get the new Chroma. I now see I need to rebase or something.

File: highlighting.go [hunk 1]
```diff
@@ -117,18 +121,22 @@ type Config struct {
 	// If WithClasses() is enabled, you can get CSS data corresponds to the style.
 	CSSWriter io.Writer
 
-	// WrapperRenderer allows you to change wrapper elements.
+	// CodeBlockOptions allows set Chroma options per code block.
+	CodeBlockOptions CodeBlockOptions
+
+	// WrapperRendererCodeBlockOptions allows you to change wrapper elements.
```
Comment (yuin) [inline highlighting.go]: Comment should be started with `WrapperRenderer`

**Comment by bep** [comment]
I have rebased and tidy'ed, but there are still many new go.sum entries. I don't know how to avoid that and get the updated Chroma.

**Comment by yuin** [comment]
@bep, OK. Changes almost no problem. I see 1 wrong comment. Could you fix it?

**Pull request merged**